### ISSUE TITLE
Issue with parsing of multiple scenario outlines in a row with tags

### ISSUE THREAD

**Issue by ucv4** [cucumber/gherkin3#104]
There appears to be an issue with Gherkin parsing multiple scenario outlines in a row where there are multiple tags at the top of the second scenario outline. I am running on version 3.1.1. See example file and walk through of code below:

Example file:

```gherkin
@ff001_tc01
@blah
Scenario Outline: blah blah
    Given <blah>
    When blah
    Then blah

    Examples:
        | blah   |
        | blah1 |
        | blah2 |

@ff001_tc02
@blahblah
Scenario Outline: blah blah
    Given blah
```

The following error gets thrown in the second scenario outline:

expected: #TagLine, #ExamplesLine, #Comment, #Empty, got 'Given blah'

The issue is that when it is processing the Examples rows and then reaches the first tag on the second Scenario Outline, it is in the match_token_at_27 function in parser.rb and hits the match_Tagline. It does the lookahead_0 and ends up putting the @blahblah piece into the context.token_queue. Then, as it continues on, it ends up processing the Scenario Outline line and then reading the @blahblah piece which is where it gets messed up since a tag cannot come after a Scenario Outline.

**Comment by brasmusson** [comment]
This problem only exist in the Ruby version of the parser, see https://github.com/cucumber/gherkin3/compare/ruby-scenario-outlines-with-tags.

**Comment by aslakhellesoy** [comment]
Thanks @ucv4 and @brasmusson. Fixed in d516035f18c5a54cd9c5a060e8bd649fbdc92469

**Comment by aslakhellesoy** [comment]
And gherkin 3.1.2 is released with this fix.
